auto heal setting of liunx web app error out with provider crash while creating through terraform

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.4

### AzureRM Provider Version

3.50.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "linux_web_app" {
  name                = "unitappservice"
  location            = "canadacentral"
  resource_group_name = "unittestrg"
  identity = {
    type = "SystemAssigned"
  }
   site_config = {
    minimum_tls_version = "1.2"
    auto_heal_enabled = true
    auto_heal_setting = {
      action = {
        action_type = "Recycle"
      }
      trigger = {
        slow_request = [{
          count    = 7
          interval = "12:25:10"
        }]
      }
    }
```


### Debug Output/Panic Output
https://gist.github.com/amaniommi39/3b44c7e960777a912f4617e91e64ec37
```shell
Error: The terraform-provider-azurerm_v3.48.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

auto heal setting should be applied on linux web app being created

### Actual Behaviour

Error: Request cancelled
│ 
│   with module.linux_web_app.azurerm_linux_web_app.linux_web_app,
│   on ../../../../web/linux_web_app/v1/linux_web_app_main.tf line 44, in resource "azurerm_linux_web_app" "linux_web_app":
│   44: resource "azurerm_linux_web_app" "linux_web_app" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.48.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

### Steps to Reproduce

run terraform init && terraform plan
run terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/helpers/linux_web_app_schema.go
+++ b/internal/services/appservice/helpers/linux_web_app_schema.go
@@ -1142,6 +1142,16 @@ func expandAutoHealSettingsLinux(autoHealSettings []AutoHealSettingLinux) *web.A
 
 	autoHeal := autoHealSettings[0]
 
+	if len(autoHeal.Actions) > 0 {
+		action := autoHeal.Actions[0]
+		result.Actions.ActionType = web.AutoHealActionType(action.ActionType)
+		result.Actions.MinProcessExecutionTime = pointer.To(action.MinimumProcessTime)
+	}
+
+	if len(autoHeal.Triggers) == 0 {
+		return result
+	}
+
 	triggers := autoHeal.Triggers[0]
 	if len(triggers.Requests) == 1 {
 		result.Triggers.Requests = &web.RequestsBasedTrigger{
@@ -1192,10 +1202,6 @@ func expandAutoHealSettingsLinux(autoHealSettings []AutoHealSettingLinux) *web.A
 		result.Triggers.StatusCodes = &statusCodeTriggers
 		result.Triggers.StatusCodesRange = &statusCodeRangeTriggers
 	}
-
-	action := autoHeal.Actions[0]
-	result.Actions.ActionType = web.AutoHealActionType(action.ActionType)
-	result.Actions.MinProcessExecutionTime = pointer.To(action.MinimumProcessTime)
 
 	return result
 }

